Lib/multiprocessing/shared_memory.py

```diff
@@ -53,13 +140,21 @@ def unlink(self):
 
 class PosixSharedMemory(_PosixSharedMemory):
 
-    def __init__(self, name, flags=None, mode=None, size=None, read_only=False):
+    def __init__(self, name, flags=None, mode=384, size=0, read_only=False):
         if name and (flags is None):
-            _PosixSharedMemory.__init__(self, name)
+            _PosixSharedMemory.__init__(self, name, mode=mode)
         else:
             if name is None:
                 name = f'psm_{os.getpid()}_{random.randrange(100000)}'
-            _PosixSharedMemory.__init__(self, name, flags=O_CREX, size=size)
+            flags = O_CREX if flags is None else flags
+            _PosixSharedMemory.__init__(
+                self,
+                name,
+                flags=flags,
+                mode=mode,
+                size=size,
+                read_only=read_only
+            )
 
         self._mmap = mmap.mmap(self.fd, self.size)
```

Comment:
This is not my territory, so take it with a grain of salt. I see that `mmap.mmap` provides some interesting arguments such as `access`, `prot`(ection), and `flags`:
https://docs.python.org/3/library/mmap.html#mmap.mmap
Depending on `mode` and `read_only` args that you provide here perhaps it would make sense to specify parameters for `mmap` in accordance (e.g. `flags=MAP_SHARED`, `prot=PROT_WRITE/READ` and `access=ACCESS_READ/WRITE).
This is a private argument so *maybe* it's not so important. Just mentioning it for completeness/correctness.

Reply:
It is a good point.  With the simpler API introduced in commit 0d3d06f7cbcb47b2dee6c027c35265795052c899, we constrain the supported behaviors to no longer offer these sorts of control.  The behavior of mmap seems to have some inconsistencies of its own across different platforms when trying to specify such modes of access -- I think this only encourages me to believe we have made the right choice with the simplified API.